app/controllers/shared/api.php

```diff
@@ -325,6 +325,44 @@
             Authorization::setDefaultStatus(false);  // Cancel security segmentation for admin users.
         }
 
+        // User verification middleware
+        $globalVerificationEnabled = System::getEnv('_APP_VERIFICATION_REQUIRED', 'disabled') === 'enabled';
+        $isConsoleProject = $project->getId() === 'console';
+        
+        if (empty($apiKey) && $globalVerificationEnabled && $isConsoleProject && !$user->isEmpty()) {
+            $currentPath = $request->getURI();
+            
+            // Endpoints that must remain accessible
+            $allowedEndpoints = [
+                '/v1/account',                    // User account operations
+                '/v1/console/variables',          // Console configuration
+                '/v1/health/version',             // Health checks
+                '/v1/account/verification',       // Email verification
+                '/v1/account/verification/phone', // Phone verification
+                '/v1/account/recovery',           // Account recovery
+                '/v1/account/sessions',           // Session management
+                '/v1/account/tokens',             // Token management
+                '/v1/account/mfa'                 // Multi-factor authentication
+            ];
+            
```

Comment:
Instead of having list of endpoints here, we can do one of 2 things:

A) Have a special `group()` for endpoints that are supposed to be blocked (might be bad idea if its too many)
B) Have a special `label()` for marking endpoint as allowed without verification

Also, this would enforce email verification on self-hosted too. It might feel like extra blocker, requiring to setup SMTP, causing people to leave the onboarding.

For that reason, I think this user verification for Console should be cloud-specific, in Cloud repository.